index.js

```diff
@@ -1,74 +1,121 @@
+            let index = 0;
+            await Promise.all(files.map(file => {
+                index += 1;
+                const jsonFile = fs.readFileSync(file);
+                const jsonContent = JSON.parse(jsonFile);
+                const sortedJsonContent = sortJson(jsonContent, {});
+                const stringContent = JSON.stringify(jsonContent);
+                const sortedStringContent = JSON.stringify(sortedJsonContent);
+
 
                 // No CI check mode. Just sort and update the given json file locally.
+                const fileName = `${file}`.bold.dim;
+                let formattedContent = JSON.stringify(sortedJsonContent, null, indent);
                 if (dryRun === false) {
-                    exec(`sort-json ${file}`, (error) => {
-                        if (error) {
-                            console.log(`error: ${error.message}`);
-                        }
-                    });
-
-                    console.info(`${file} is sorted alphabetically!`);
-                    return 0;
+                    if (force && stringContent === sortedStringContent) {
+                        fs.writeFileSync(file, formattedContent)
+                        console.info(`[${index}] ${fileName} sorted (forced)`.blue);
+                        unsortedFiles.push(file);
+                    } else if (stringContent !== sortedStringContent) {
+                        console.info(`[${index}] ${fileName} sorted`.yellow);
+                        fs.writeFileSync(file, formattedContent)
+                        unsortedFiles.push(file);
+                    } else if (!unsortedOnly) {
+                        console.info(`[${index}] ${fileName} is already sorted`.green);
+                    }
+                    return;
                 }
 
                 // CI check mode. Compare sorted file and local file.
-                if (JSON.stringify(jsonContent) !== JSON.stringify(sortedJsonContent)) {
-                    throw new Error(`${file} not sorted alphabetically`);
+                if (stringContent !== sortedStringContent) {
+                    console.error(`[${index}] ${fileName} not sorted`.red);
+                    unsortedFiles.push(file);
+                } else if (!unsortedOnly) {
+                    console.info(`[${index}] ${fileName} is sorted`.green);
                 }
+            }));
 
-                console.info(`${file} is already sorted alphabetically`);
-                return 0;
-            });
 
-            return 0;
+            if (unsortedFiles.length > 0) {
+                if (dryRun) {
+                    console.info(`\n${unsortedFiles.length} of ${files} file(s) not sorted`.red);
+                     process.exit(1);
+                }
+                console.info(`\n${unsortedFiles.length} files sorted`.yellow.bold);
+                process.exit(0);
+            }
+
+            console.info(`\nAll files (${files.length}) are sorted`.green.bold);
```

Comment:
Maybe a bit of over engineering but I would suggest a different text for normal, forced, or dry-run

Reply:
Yes, feels a bit too much. Makes code less readable). Had this before 😁